### PR TITLE
[4.0] Ignore template css

### PR DESCRIPTION
**Pull request by Bakual**
After running npm install, you currently get a bunch of modified template CSS files and one untracked one.

### Summary of Changes
This PR removes the compiled CSS files from the repo and adds them to .gitignore.

### Testing Instructions
Apply PR and run `npm install --unsafe-perm`

### Expected result
The files are created and the template works
Git shows no changed or untracked CSS files for templates.

### Actual result
Everything works, but Git complains.

### Documentation Changes Required
None

### CODE REVIEW

**Comment by brianteeman** [comment]
I thought we were keeping those template CSS for now to help with testing so you don't have to do an npm install everyone you need to test a quick CSS change in a pr. 

At least that was my understanding

**Comment by Bakual** [comment]
I see changed files everytime I run npm install on the repo. Either because they are outdated or because my env compiles them different than when it was commited - I don't know. It's confusing. 😕 

I've asked in the maintainer chat if it is intentional and was told to remove them. So no clue 😄

**Comment by brianteeman** [comment]
Maybe I misunderstood what @wilsonge said

**Comment by dgrammatiko** [comment]
@Bakual @brianteeman the conversation is about moving or not the sources in build/whatever and me proposing to move also the compiled files in media/whatever. 
Maybe one reason that I didn't mention before is that having all the static assets under one folder makes the implementation of a service worker for them a 5 line task...

**Comment by brianteeman** [comment]
This is the comment from @wilsonge that has perhaps confused me https://github.com/joomla/joomla-cms/pull/21306#issuecomment-410668006

**Comment by Bakual** [comment]
Not sure why he wants to make that dependant on a decision of where those files live in the end.
Imho we either remove all compiled files from the repo or we don't. But having the compiled template CSS files in the repo while all other compiled CSS files are gone makes no sense to me.

Adding them to .gitignore is really not that hard. It's not like we're talking about 100 files. And once the sources are moved to whatever place, we can probably just .gitignore the templates CSS folder instead of single files like now.

**Comment by Bakual** [comment]
Rebased, solved conflicts and made it more complete.

File: .gitignore [hunk 1]
```diff
@@ -22,6 +22,15 @@
 /.php_cs.cache
 /media
 
+# Template CSS files generated by NPM.
+/administrator/templates/atum/css
+/installation/template/css/template.min.css
+/installation/template/css/template.css
```
Comment (infograf768) [inline .gitignore]: what about `installation/template/css/template_rtl.css` ?

Comment (infograf768) [reply .gitignore]: Is it because we do not have a scss file for it?

Comment (Bakual) [reply .gitignore]: I think so. The rtl file apparently isn't generated by NPM. I've tested by deleting and running `npm run build:css`

Comment (brianteeman) [reply .gitignore]: then thats a bug

Comment (Bakual) [reply .gitignore]: Probably, but not one I'm capable of solving (yet). Haven't looked at SCSS files yet.

**Comment by Bakual** [comment]
Rebased again to take care of https://github.com/joomla/joomla-cms/pull/21810

File: .gitignore (hunk 1 above)
Comment (Bakual) [reply .gitignore]: Fixed by JM 👍

**Comment by wilsonge** [comment]
> Not sure why he wants to make that dependant on a decision of where those files live in the end

I thought we were going to have that conversation almost immediately as everything else was happening. but it never happened. So doing this now is fine i guess.

**Pull request merged**